ex2_hal/iris/hardware_interface/source/iris_bootloader_cmds.c

```diff
@@ -97,14 +98,14 @@ int iris_write_page(uint32_t flash_addr, uint8_t *buffer) {
     static uint8_t packet[WRITE_PACKET_LENGTH];
     uint8_t flash_mem_checksum = 0x00;
     uint8_t data_checksum = 0x00;
-    int ret = 0x00;
+    int ret = 0;
     uint8_t rx_data;
 
     /* First I2C transaction (2 bytes) */
     packet[0] = OPC_WRITE;
     packet[1] = N_OPC_WRITE;
-    iris_write_packet(packet, 2);
-    iris_read_packet(&rx_data, 1);
+    ret += iris_write_packet(packet, 2);
```

Comment:
There's no guarantee subsequent commands worked, shouldn't ret be checked every time and return on failure?

Reply:
The granularity of error handling for Iris firmware updates is not required. We want to only capture if the overarching API fails or passes. Both `iris_write_packet` and `iris_read_packet` are wrappers for i2c handler, if one of them fails then it is guaranteed that both will fail. Also the subsequent commands are actually derived directly from the [STM32 manual ](file:///home/jenish/Downloads/an4221-i2c-protocol-used-in-the-stm32-bootloader-stmicroelectronics.pdf)

Reply:
That is fine, we don't need granularity for when it failed, but the OBC shouldn't continue doing stuff if it did fail, even if the stuff is harmless

Reply:
I guess that is right indeed. Have a look at the new changes